packages/vite/src/node/plugins/resolve.ts

```diff
@@ -239,7 +239,10 @@ export function resolvePlugin(resolveOptions: InternalResolveOptions): Plugin {
       }
 
       // absolute fs paths
-      if (path.isAbsolute(id) && (res = tryFsResolve(id, options))) {
+      if (
+        isNonDriveRelativeAbsolutePath(id) &&
+        (res = tryFsResolve(id, options))
+      ) {
```

Comment:
Even if the condition above is fixed, this is still needed.
`/foo.svelte` will be handled as `C:/foo.svelte` inside `tryFsResolve` by `fs.statSync` and others.